clang/tools/libclang/CXLoadedDiagnostic.cpp

```diff
@@ -243,31 +243,48 @@ class DiagLoader : serialized_diags::SerializedDiagnosticReader {
   }
 
   CXDiagnosticSet load(const char *file);
+  CXDiagnosticSet load(std::unique_ptr<llvm::MemoryBuffer> Buffer);
```

Comment:
Does this need to take ownership of the MemoryBuffer? Could we pass `llvm::MemoryBufferRef` instead?

Reply:
Good idea, changed to `llvm::MemoryBufferRef` here: https://github.com/apple/llvm-project/pull/6566/commits/2bd7ad87991aafdc4c97b8ed6673bbc486c97d2d